python/plugins/processing/core/ProcessingConfig.py

```diff
@@ -83,6 +84,10 @@ def initialize():
             ProcessingConfig.tr('Use only selected features'), True))
         ProcessingConfig.addSetting(Setting(
             ProcessingConfig.tr('General'),
+            ProcessingConfig.IGNORE_NULL_GEOMS,
+            ProcessingConfig.tr('Ignore features with null geometries'), True))
```

Comment:
I'd have thought this setting is very algorithm dependent. Eg Extract by Location - > would expect to ignore null geometries. But an algorithm like Extract by Attribute should never ignore null geometries. 

Maybe a better approach would be:
1. make sure all qgis algs correctly handle null geometries - eg, copy the feature with null geometry unless (like Extract by Location) it makes no sense to do so.
2. add a new algorithm "remove features with no geometry". Then the user is in full control of how this is handled.